src/shared/Core/ProcessManager.cs

```diff
@@ -50,9 +57,53 @@ public virtual ChildProcess CreateProcess(string path, string args, bool useShel
         return CreateProcess(psi);
     }
 
-
     public virtual ChildProcess CreateProcess(ProcessStartInfo psi)
     {
         return new ChildProcess(Trace2, psi);
     }
+
+    /// <summary>
+    /// Create a TRACE2 "session id" (sid) for this process.
+    /// </summary>
+    public static void CreateSid()
+    {
+        Sid = Environment.GetEnvironmentVariable(SidEnvar);
+
+        if (!string.IsNullOrEmpty(Sid))
+        {
+            Sid = $"{Sid}/{Guid.NewGuid():D}";
+        }
+        else
+        {
+            // We are the root process; create our own 'root' SID
+            Sid = Guid.NewGuid().ToString("D");
+        }
+
+        Environment.SetEnvironmentVariable(SidEnvar, Sid);
+        Depth = GetProcessDepth();
```

Comment:
nit: you can save a few cycles if you assign Depth inside the above "if", since we already know the answer in the null case.